src/tracker/inflights.rs

```diff
@@ -122,6 +127,23 @@ impl Inflights {
         self.count = 0;
         self.start = 0;
     }
+
+    /// free buffer memory by reassignment
+    #[inline]
+    pub fn free_mem(&mut self) {
+        let mut v = Vec::with_capacity(self.count);
+        for i in 0..self.count {
+            v.push(self.buffer[self.start + i])
+        }
+        self.buffer = v;
+        self.start = 0;
+    }
+
+    /// reallocate buffer memory
+    #[inline]
+    pub fn reallocate(&mut self) {
```

Comment:
Progress should reallocate when necessary.

Reply:
Currently, buffer will be reallocated when you call this function, what is other situation when we need reallocation?